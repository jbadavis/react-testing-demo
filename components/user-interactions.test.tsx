import {render, screen} from '@testing-library/react';
import {userEvent} from '@testing-library/user-event';
import UserInteractions from './user-interactions';

describe('UserInteractions', () => {
  let container: HTMLElement;

  describe('when button is not clicked', () => {
    beforeEach(() => {
      container = render(<UserInteractions />).container;
    });

    test('render the correct button copy', () => {
      expect(screen.getByRole('button').innerHTML).toBe('Show');
    });

    test('not render the scary element', () => {
      expect(screen.queryByText('ğŸ‘» BOO! ğŸ‘»')).toBeNull();
    });

    test('render correctly', () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe('when text is inputted', () => {
    beforeEach(async () => {
      const user = userEvent.setup();

      container = render(<UserInteractions />).container;

      await user.type(screen.getByTestId('an-input'), 'foo bar baz');
    });

    test('render the correct copy', () => {
      expect(screen.getByTestId('user-output').innerHTML).toContain(
        'foo bar baz',
      );
    });

    test('render correctly', () => {
      expect(container).toMatchSnapshot();
    });
  });

  describe('when the button is clicked', () => {
    beforeEach(async () => {
      const user = userEvent.setup();

      container = render(<UserInteractions />).container;

      await user.click(screen.getByRole('button'));
    });

    test('render the correct button copy', () => {
      expect(screen.getByRole('button').innerHTML).toBe('Hide');
    });

    test('render the scary element', () => {
      expect(screen.queryByText('ğŸ‘» BOO! ğŸ‘»')).toBeInTheDocument();
    });

    test('render correctly', () => {
      expect(container).toMatchSnapshot();
    });
  });
});
