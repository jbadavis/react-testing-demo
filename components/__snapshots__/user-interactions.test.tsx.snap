// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInteractions when button is not clicked render correctly 1`] = `
<div>
  <h2
    class="chakra-heading css-jmbq56"
  >
    User Interactions
  </h2>
  <div
    class="chakra-stack css-1sam6cv"
  >
    <p
      class="chakra-text css-0"
    >
      User interactions are often part of the public interface of our UI our component. In order to test them we need to simulate user interactions, such a clicking a button or typing into an input field.
    </p>
    <h3
      class="chakra-heading css-0"
    >
      Some Typing
    </h3>
    <input
      class="chakra-input css-0"
      data-testid="an-input"
      placeholder="Type something or else"
    />
    <p
      class="chakra-text css-vmpf5w"
      data-testid="user-output"
    >
      <strong>
        You are typing:
      </strong>
       
    </p>
    <h3
      class="chakra-heading css-0"
    >
      A Click
    </h3>
    <button
      class="chakra-button css-1cthc8r"
      type="button"
    >
      Show
    </button>
  </div>
</div>
`;

exports[`UserInteractions when text is inputted render correctly 1`] = `
<div>
  <h2
    class="chakra-heading css-jmbq56"
  >
    User Interactions
  </h2>
  <div
    class="chakra-stack css-1sam6cv"
  >
    <p
      class="chakra-text css-0"
    >
      User interactions are often part of the public interface of our UI our component. In order to test them we need to simulate user interactions, such a clicking a button or typing into an input field.
    </p>
    <h3
      class="chakra-heading css-0"
    >
      Some Typing
    </h3>
    <input
      class="chakra-input css-0"
      data-testid="an-input"
      placeholder="Type something or else"
    />
    <p
      class="chakra-text css-vmpf5w"
      data-testid="user-output"
    >
      <strong>
        You are typing:
      </strong>
       
      foo bar baz
    </p>
    <h3
      class="chakra-heading css-0"
    >
      A Click
    </h3>
    <button
      class="chakra-button css-1cthc8r"
      type="button"
    >
      Show
    </button>
  </div>
</div>
`;

exports[`UserInteractions when the button is clicked render correctly 1`] = `
<div>
  <h2
    class="chakra-heading css-jmbq56"
  >
    User Interactions
  </h2>
  <div
    class="chakra-stack css-1sam6cv"
  >
    <p
      class="chakra-text css-0"
    >
      User interactions are often part of the public interface of our UI our component. In order to test them we need to simulate user interactions, such a clicking a button or typing into an input field.
    </p>
    <h3
      class="chakra-heading css-0"
    >
      Some Typing
    </h3>
    <input
      class="chakra-input css-0"
      data-testid="an-input"
      placeholder="Type something or else"
    />
    <p
      class="chakra-text css-vmpf5w"
      data-testid="user-output"
    >
      <strong>
        You are typing:
      </strong>
       
    </p>
    <h3
      class="chakra-heading css-0"
    >
      A Click
    </h3>
    <button
      class="chakra-button css-1cthc8r"
      type="button"
    >
      Hide
    </button>
    <h3
      class="chakra-heading css-0"
    >
      ðŸ‘» BOO! ðŸ‘»
    </h3>
  </div>
</div>
`;
